### PR TITLE
[DEVHUB-1380]: Sign Up Modal - Markup & Styling

### PR DESCRIPTION
**Pull request by justinmckenzie**

### CODE REVIEW

File: src/components/form/form.tsx [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+        >
+          <FormGeneric
+            inverse
+            customStyles={{
+              width: ['100%', '440px', null, '100%'],
+            }}
+          >
+            <Field
+              component="text-input"
+              // validators={[isRequired(), minLength(3), maxLength(10)]}
```
Comment (gfunk1230) [inline src/components/form/form.tsx:30]: Do we intentionally want to leave these here commented?

File: src/components/form/form.tsx [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+          title="Join the MongoDB Educator Community"
+          onClose={onClose}
+        >
+          <FormGeneric
+            inverse
+            customStyles={{
+              width: ['100%', '440px', null, '100%'],
+            }}
+          >
+            <Field
```
Comment (gfunk1230) [inline src/components/form/form.tsx:28]: Nitpick, but would it be better to use a map to loop over each `Field` component? So something like so:

```
const fields = [{ name: "lastname", label: "Last Name", component: "text-input" },{ name: "email", label: "email", component: "text-input" }];

fields.map(({ name, label, component }) => (
	<Field component={component} name={name} label={label} />
))
```

File: src/components/form/styles.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import { ThemeUICSSObject } from 'theme-ui';
+
+const FormModalWrapper: ThemeUICSSObject = {
+  display: 'flex',
```
Comment (gfunk1230) [inline src/components/form/styles.ts:4]: Super nitpick, but we try to alphabetize the CSS :-/

**Review by gfunk1230 — APPROVED** [review]
Left some comments but other than that LGTM!

File: src/components/form/form.tsx (hunk 1 above)
Comment (justinmckenzie) [reply src/components/form/form.tsx:30]: left them commented for now, didn't know what req's would look like for form validation. Was gonna address it during integration

File: src/components/form/form.tsx (hunk 2 above)
Comment (justinmckenzie) [reply src/components/form/form.tsx:28]: Had a similar thought, was going to wait for integration to see what the data looked like but will just include that in this branch.

File: src/components/form/styles.ts (hunk 3 above)
Comment (justinmckenzie) [reply src/components/form/styles.ts:4]: ill clean up any touched `styles` files here with the alphabetizing...will raise a separate PR for the other ones.

File: src/components/form/form.tsx [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+    name: 'institutionType',
+    label: 'Institution Type',
+    component: 'select',
+    options: institutionTypes,
+  },
+  {
+    name: 'location',
+    label: 'Country',
+    component: 'select',
+    options: ['United States of America', 'Canada', 'Mexico'], // TODO: proper list of these - fetched data?
```
Comment (mcolella14) [inline src/components/form/form.tsx:60]: Ah yeah so I was looking into this a while ago and https://github.com/esosedi/3166 seems like a solid source for these. I know the current devhub academia page is using https://www.npmjs.com/package/react-select-country-list, which is a little outdated but would give us parity?

But regardless I think we should pull from a list. May have to prop drill these down, or use https://www.npmjs.com/package/next-plugin-preval to populate the list at build time and just import it.

File: src/lib/registration.ts [hunk 5]
```diff
@@ -70,7 +70,7 @@ export const validateRegistrationBody = (body: any) => {
 
   if (!teachingStatuses.includes(body.teachingStatus)) {
     throw Error(
-      `${body.eachingStatus} is not a valid value for teachingStatus`
+      `${body.teachingStatus} is not a valid value for teachingStatus`
```
Comment (mcolella14) [inline src/lib/registration.ts:73]: Ah sorry about that lol, thanks for fixing it.

File: src/components/form/form.tsx [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+  {
+    name: 'email',
+    label: 'Email',
+    component: 'text-input',
+  },
+  {
+    name: 'jobFunction',
+    label: 'Your Job Function',
+    component: 'select',
+    options: jobFunctions,
```
Comment (mcolella14) [inline src/components/form/form.tsx:37]: So on the current form, when the "Other" option is selected from this list, a new form field appears asking the user to specify. I would confirm with Liz, but at least to have functional parity we would need something like this.

**Review by mcolella14 — COMMENTED** [review]
Looks great! In terms of validation, I think we should start by following suit with the current implementation. Here is [the validation code that currently exists](https://github.com/mongodb/devhub/blob/staging/src/components/dev-hub/academia-sign-up-form.js#L107), and [the existing educators page](https://www.mongodb.com/developer/academia/educators/) if you want to experiment with the form.

File: src/components/form/form.tsx (hunk 6 above)
Comment (justinmckenzie) [reply src/components/form/form.tsx:37]: @mcolella14 thanks for the feedback. I am going to set up some time with Liz this week to go over everything on Forms, I will bring this up along with your points on field validation.

**Pull request merged**